spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/quartz/TracingJobListener.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.quartz;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import brave.Span;
+import brave.Tracer;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.JobListener;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * {@link org.quartz.JobListener JobListener} that will wrap a span around quartz jobs
+ * when they start and finish.
+ *
+ * @author Branden Cash
+ */
+@Component
+public class TracingJobListener implements JobListener {
+
+	private final Map<String, Span> jobSpans = new HashMap<>();
+
+	@Autowired
+	private Tracer tracer;
+
+	@Override
+	public String getName() {
+		return getClass().getName();
+	}
+
+	@Override
+	public void jobToBeExecuted(JobExecutionContext context) {
+		String name = String.format("Job: %s; Trigger: %s",
+				context.getTrigger().getJobKey(), context.getTrigger().getKey());
+		Span span = tracer.nextSpan().name(name);
+		jobSpans.put(context.getFireInstanceId(), span);
```

Comment:
Instead of using a map in this class I would use the `put` method on the `JobExecutionContext` to attach the objects you need later to the quartz context

Reply:
agree. map is dodgiest option, can accidentally leak etc. use as last resort

Reply:
Very good suggestion :D thank you

Reply:
Code has been updated. Please review and resolve the conversation if the concern is resolved.